File: backend/app/schemas/invoice.py
from pydantic import BaseModel
from datetime import date
from decimal import Decimal



class InvoiceLineItem(BaseModel):

    id: int
    work_date: date
    project_name: str
    description: str | None
    hours: Decimal
    hourly_rate: Decimal
    amount: Decimal


class InvoicePreview(BaseModel):

    client_id: int
    client_name: str
    start_date: date
    end_date: date
    line_items: list[InvoiceLineItem]
    total_hours: Decimal
    total_amount: Decimal

File: frontend/src/types/invoice.ts

export interface InvoiceLineItem {

    id: number
    work_date: string
    project_name: string
    description: string | null
    hours: number
    hourly_rate: number
    amount: number
}

export interface InvoicePreview {

    client_id: number
    client_name: string
    start_date: string
    end_date: string
    line_items: InvoiceLineItem[]
    total_hours: number
    total_amount: number
}